### PR TITLE
[TE] rootcause poc - hook up rca v2 backend

### PR DESCRIPTION
**Pull request by apucher**
The backend now supports long-form URNs (e.g. with filters) for metric analysis for both input and output. This PR adapts the frontend to enable full long-form support in metric analysis and event analysis. Also fixes the most egregious sort order and length-related display issues.

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-legend/component.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default Ember.Component.extend({
     'validUrns',
     function () {
       const { entities, validUrns } = this.getProperties('entities', 'validUrns');
-      return validUrns.filter(urn => entities[stripTail(urn)]).reduce((agg, urn) => { agg[urn] = entities[stripTail(urn)].color; return agg; }, {});
+      return validUrns.filter(urn => entities[urn]).reduce((agg, urn) => { agg[urn] = entities[urn].color; return agg; }, {});
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-legend/component.js]: For readability's sake, please span over multiple lines when chaining methods.

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js [hunk 2]
```diff
@@ -55,27 +50,27 @@ export default Ember.Component.extend({
       const metricUrns = filterPrefix(Object.keys(entities), ['thirdeye:metric:']);
       let output = [];
 
-      if (sortProperty == ROOTCAUSE_METRICS_SORT_PROPERTY_METRIC) {
-        output = metricUrns.map(urn => [metrics[urn], urn]).sort().map(t => t[1]);
+      if (sortProperty === ROOTCAUSE_METRICS_SORT_PROPERTY_METRIC) {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js]: I feel that a `switch case` might be cleaner here

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-legend/component.js (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-legend/component.js]: fixed

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js (hunk 2 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js]: fixed

**Pull request merged**